src/libcore/iter/traits/iterator.rs

```diff
@@ -201,13 +201,13 @@ pub trait Iterator {
 
     /// Consumes the iterator, counting the number of iterations and returning it.
     ///
-    /// This method will evaluate the iterator until its [`next`] returns
-    /// [`None`]. Once [`None`] is encountered, `count()` returns one less than the
-    /// number of times it called [`next`]. Note that [`next`] has to be called at
-    /// least once even if the iterator does not have any elements.
+    /// This method will call [`next`] repeatedly until [`None`] is encountered,
+    /// returning the number of times it saw [`Some`]. Note that [`next`] has to be
+    /// called at least once even if the iterator does not have any elements.
```

Comment:
> Note that [`next`] has to be called at least once even if the iterator does not have any elements.

I added this sentence earlier but now I doubt that it should be there, as it probably doesn't help in any way.

Reply:
It doesn't seem super necessary, but I don't feel too strongly about it.

Reply:
Alright, then I'm okay with merging this.